src/one/nio/serial/GeneratedSerializer.java

```diff
@@ -38,13 +38,20 @@
     static final AtomicInteger migratedFields = new AtomicInteger();
     static final AtomicInteger renamedFields = new AtomicInteger();
     static final AtomicInteger unsupportedFields = new AtomicInteger();
+    static final Set<Class<?>> generationInProcess = new ConcurrentSkipListSet<Class<?>>(new Comparator<Class<?>>() {
+        @Override
+        public int compare(Class<?> o1, Class<?> o2) {
+            return o1.getName().compareTo(o2.getName());
+        }
+    });
```

Comment:
Why not just a regular ConcurrentHashMap? Also, please mind that classes from different ClassLoaders may have the same names.